src/persist/src/file.rs

```diff
@@ -19,6 +20,8 @@ use fail::fail_point;
 
 use tokio::fs::{self, File, OpenOptions};
 use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::sync::Mutex;
+use walkdir::{DirEntry, WalkDir};
```

Comment:
i'm generally pretty hesitant to add new deps to persist if it's at all reasonable to avoid. rust build times can get out of hand fast. what does this get us over https://doc.rust-lang.org/std/fs/struct.ReadDir.html?

somewhat related, after looking at the complexity of this, I'm temped to walk back my suggestion to store these as actual hierarchical dirs in FileBlob. instead, maybe we should just escape the `/`s and store it all flat. the fileblob impl is only for dev, so we shouldn't have to worry about limits of how many files can be in a dir. it also seems like a better semantic match for Blob, which doesn't have any notion of structured hierarchies, just prefixes. thoughts?

Reply:
I've flipflopped on this one a few times... my first reaction was that flat files was definitely the way to go to match S3. Then that we should use directories to make things like listing files within a shard super easy. This implementation seems fragile though, so I'd be happy to switch it back